fake-vm-component-scripts/fake-vm/launch.py

```diff
@@ -10,25 +10,30 @@
 
 instances = []
 
-def fix_config_keys(source):
-    for k, v in source:
-        yield (k, v) if '.' not in k else (k.split('.')[1], v)
+config_keys = {
+    k: k.split('.')[1] for k in {
+        'configuration.name',
+        'configuration.address',
+        'configuration.login',
+        'configuration.failure',
```

Comment:
Well, ideally this name should be obtained from the instance name, but we don't expose instance names on scriptable components so I passed it through the configuration. Why this may be useful: imagine that the user has launched a VM in the platform, and then it goes down together with its database. A new platform is started up, and discovers the VM. Since it has a name, it is created through discovery in the portal. If it hadn't had one, it won't be created.

And, after all, I imagine that a name is a very valid parameter for a VM. Amazon VMs, for example, may store a user-friendly name in its tag.